lms/djangoapps/courseware/tests/test_access.py

```diff
@@ -45,9 +57,98 @@
     seed_milestone_relationship_types,
 )
 
+from lms.djangoapps.ccx.models import CustomCourseForEdX
+
 # pylint: disable=protected-access
 
 
+class CoachAccessTestCaseCCX(SharedModuleStoreTestCase, LoginEnrollmentTestCase):
+    """
+    Test if user is coach on ccx.
+    """
+    MODULESTORE = TEST_DATA_SPLIT_MODULESTORE
+
+    @classmethod
+    def setUpClass(cls):
+        """
+        Set up course for tests
+        """
+        super(CoachAccessTestCaseCCX, cls).setUpClass()
+        cls.course = CourseFactory.create()
+
+    def setUp(self):
+        """
+        Set up tests
+        """
+        super(CoachAccessTestCaseCCX, self).setUp()
+
+        # Create ccx coach account
+        self.coach = AdminFactory.create()
+        self.client.login(username=self.coach.username, password="test")
+
+        # assign role to coach
+        role = CourseCcxCoachRole(self.course.id)
+        role.add_users(self.coach)
```

Comment:
@singingwolfboy i can add it as you are asking , but this will be out of scope of this PR, 
@2nd thing is this logic is used everywhere in edX. I have to add this line then i need to borrow code from roles.py the link i just sent you.